### PR TITLE
Readme / Comparison Updates

### LINKED ISSUE #474: Update README
We have to update the README for the new algos that are in the repo. We also have to update the benchmark results.

@gramhagen could you do VW?

I will do FastAI and NCF

The list of the new algos that we have is:
- [x]  VW
- [x]  NCF
- [x]  FastAI
- [x]  xDeepFM
- [x]  DKN
- [x]  RBM

### PR DESCRIPTION
**Pull request by nikhilrj**
### Description
- Update Readme to highlight new 0.2 algos
- Update comparison notebook to run new notebooks

### Motivation and Context
- So a user can easily find all algorithms available in the repo

### Releated Issues
Fixes #474 

### Checklist:

- [ ] My code follows the code style of this project, as detailed in our [contribution guidelines](../CONTRIBUTING).
- [ ] I have added tests.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: notebooks/03_evaluate/comparison.ipynb [hunk 1]
```diff
@@ -373,16 +466,16 @@
...
-       "      <td>0.049102</td>\n",
-       "      <td>0.050849</td>\n",
-       "      <td>0.018680</td>\n",
-       "      <td>0.948764</td>\n",
-       "      <td>0.742875</td>\n",
-       "      <td>0.283734</td>\n",
-       "      <td>0.288698</td>\n",
-       "      <td>3.484409</td>\n",
-       "      <td>10.361868</td>\n",
+       "      <td>0.005232</td>\n",
```
Comment (nikhilrj) [inline notebooks/03_evaluate/comparison.ipynb]: @gramhagen would it make sense to include results from a VW algo here?

File: README.md [hunk 2]
```diff
@@ -52,15 +52,39 @@ We provide several notebooks to show how recommendation algorithms can be design
 
 - The [Operationalizion Notebook](notebooks/05_operationalize) demonstrates how to deploy models in production systems.
 
+
+The Quick-Start and Modeling notebooks showcase how to utilize the following algorithms to build a recommender system:
+
+**Algorithms**
+
+| Algorithm | Environment | Description |
```
Comment (nikhilrj) [inline README.md]: Ideally in the description we should put 1 sentence overviews of the strengths and weaknesses of the algorithm, and where the algorithm should be used. This description should NOT just describe the algorithm and dataset we're using it on, but also describe when and where a user should use this algorithm (where does it shine?),

File: README.md [hunk 3]
```diff
@@ -52,15 +52,39 @@ We provide several notebooks to show how recommendation algorithms can be design
...
+
+**Algorithms**
+
+| Algorithm | Environment | Description | 
+| --- | --- | --- |
+| **`Classic Recommenders`** |
+| [Surprise/Singular Value Decomposition (SVD)](notebooks/00_quick_start/sar_single_node_movielens.ipynb) | Python |  | 
+| [Alternating Least Squares (ALS)](notebooks/00_quick_start/als_pyspark_movielens.ipynb) | Spark |  |
+| **`Microsoft Recommenders`** |
+| [Smart Adaptive Recommendations (SAR)](notebooks/00_quick_start/sar_single_node_movielens.ipynb) | Python / Spark | |
```
Comment (miguelgfierro) [inline README.md]: In the setup, we were referring to the environments as:
* Python CPU
* Python GPU
* PySpark
we should have consistency here

File: README.md (hunk 2 above)
Comment (miguelgfierro) [reply README.md]: I think that makes sense

File: README.md [hunk 4]
```diff
@@ -52,15 +52,39 @@ We provide several notebooks to show how recommendation algorithms can be design
...
+| [Vowpal Wabbit Family (VW)](notebooks/02_model/vowpal_wabbit_deep_dive.ipynb) | Python / Online | |
+| **`Deep Learning`** |
+| [eXtreme Deep Factorization Machine (xDeepFM)](notebooks/00_quick_start/xdeepfm.ipynb) | Python / GPU | | 
+| [Deep Knowledge-Aware Network (DKN)](notebooks/00_quick_start/dkn.ipynb) | Python / GPU | | 
+| [Neural Collaborative Filtering (NCF)](notebooks/00_quick_start/ncf_movielens.ipynb) | Python / GPU | | 
+| [Restricted Boltzmann Machines (RBM)](notebooks/00_quick_start/rbm_movielens.ipynb) | Python / GPU | | 
+| [FastAI](notebooks/00_quick_start/fastai_recommendation.ipynb)  | Python / GPU | |
```
Comment (miguelgfierro) [inline README.md]: I think we should rename FastAI to FastAI EmbeddingDotBias https://docs.fast.ai/collab.html#EmbeddingDotBias, which is the algo that we are using. I'm going to make a modification of the fastai notebook to make this more clear

File: notebooks/00_quick_start/fastai_recommendation.ipynb [hunk 5]
```diff
@@ -27,12 +27,12 @@
      "name": "stdout",
```
Comment (miguelgfierro) [inline notebooks/00_quick_start/fastai_recommendation.ipynb]: do we need to run all these notebooks again? In the fastai one, we are not using GPU

File: notebooks/03_evaluate/comparison.ipynb (hunk 1 above)
Comment (miguelgfierro) [reply notebooks/03_evaluate/comparison.ipynb]: +1 to that

File: README.md [hunk 6]
```diff
@@ -52,15 +52,39 @@ We provide several notebooks to show how recommendation algorithms can be design
...
+The Quick-Start and Modeling notebooks showcase how to utilize the following algorithms to build a recommender system:
+
+**Algorithms**
+
+| Algorithm | Environment | Description | 
+| --- | --- | --- |
+| **`Classic Recommenders`** |
+| [Surprise/Singular Value Decomposition (SVD)](notebooks/00_quick_start/sar_single_node_movielens.ipynb) | Python |  | 
+| [Alternating Least Squares (ALS)](notebooks/00_quick_start/als_pyspark_movielens.ipynb) | Spark |  |
+| **`Microsoft Recommenders`** |
```
Comment (miguelgfierro) [inline README.md]: One problem I see with the classification of classic, Microsoft and DL recommenders is that xDeepFM and DKN are also MS recommenders, but they are in DL...

File: README.md [hunk 7]
```diff
@@ -52,15 +52,39 @@ We provide several notebooks to show how recommendation algorithms can be design
...
 
 **Preliminary Comparison**
 
 | Algo | MAP | nDCG@k | Precision@k | Recall@k | RMSE | MAE | R<sup>2</sup> | Explained Variance | 
 | --- | --- | --- | --- | --- | --- | --- | --- | --- | 
-| ALS | 0.002020 | 0.024313 | 0.030677 | 0.009649 | 0.860502 | 0.680608 | 0.406014 | 0.411603 | 
-| SAR | 0.064013 | 0.308012 | 0.277215 | 0.109292 | N/A | N/A | N/A | N/A | 
-| SVD | 0.010915 | 0.102398 | 0.092996 | 0.025362 | 0.888991 | 0.696781 | 0.364178 | 0.364178 | 
+| [ALS](notebooks/00_quick_start/als_pyspark_movielens.ipynb) | 0.002020 | 0.024313 | 0.030677 | 0.009649 | 0.860502 | 0.680608 | 0.406014 | 0.411603 |
```
Comment (miguelgfierro) [inline README.md]: guys, does it make sense to add here the training and testing time? There should be an important difference between VW and RBM for example

Comment (gramhagen) [reply README.md]: I think having train time would be good, test time might not be as useful if different metrics are being applied, or if the majority of the time is really just due to slow evaluation code, but if we can get to a equitable way to demonstrate evaluation time that would be good too

Comment (miguelgfierro) [reply README.md]: ok guys, I'm going to work on getting the times of all algos and will do a commit here

Comment (miguelgfierro) [reply README.md]: FYI https://github.com/Microsoft/Recommenders/issues/464#issuecomment-461037795

File: README.md (hunk 6 above)
Comment (nikhilrj) [reply README.md]: Moved xDeepFM and DKN to MS (still open to how to best show this)

File: notebooks/03_evaluate/comparison.ipynb (hunk 1 above)
Comment (gramhagen) [reply notebooks/03_evaluate/comparison.ipynb]: is the current plan to update the comparison notebook so we have apples to apples for all algos?
if you need current results I can provide them

File: README.md (hunk 7 above)
Comment (WessZumino) [reply README.md]: @gramhagen I have interpreted test time as the time needed to return recommendation on the test set rather then the time needed to evaluate the metrics. In the former case, test time would dependent only on the number of returned recommendation, i.e. 10 elements against all of them I think. The former definition is as you said, it depends on the evaluation algo. Now I am not sure which definition we are using :)

Comment (WessZumino) [reply README.md]: @miguelgfierro just a small note: I noticed that gpu algos needs more time to run the first time they are instantiated. In the case of RBM, the first time you run it takes considerably longer to run than the next n+1 times. I think this is due to TF initializing the GPU.

Concerning the reported metrics, I noticed  that they are lower then the one reported in the notebooks. Is there a reason for that?

**Pull request merged**